userapi/src/main/java/fk/prof/userapi/util/StackLineParser.java

```diff
@@ -0,0 +1,122 @@
+        }
+    } ;
+
+    public static String convertJVMTypeSignToJava(String text){
+        StringBuilder sb = new StringBuilder();
+        int textIndex = 0;
+        int sbLength = 0;
+        text = text.replace('/','.');
+        for(JVMTypeSignatureParser parser : parsers){
+            int res = parser.parse(sb, text, textIndex);
+            if(res == -1){
+                return sb.substring(0, sbLength) + text.substring(textIndex);
+            }
+            textIndex = res;
+            sbLength = sb.length();
+        }
+        return sb.toString();
+    }
+
+
+
+    @FunctionalInterface
+    private interface JVMTypeSignatureParser{
+        int parse(StringBuilder sb, String text, int textIndex);
+    }
+
+    private static int parseTerminalType(StringBuilder sb, String text, int textIndex){
+        if(textIndex >= text.length()) return -1;
+        char currChar = text.charAt(textIndex);
+        if(currChar == '['){
+            int nextIndex = parseTerminalType(sb, text, textIndex + 1 );
+            sb.append("[]");
+            return nextIndex;
+        }else if(currChar == 'L'){
+            int endIndex = text.indexOf(';', textIndex);
+            if(endIndex == -1) return -1;
+            sb.append(text, textIndex+1, endIndex);
+            return endIndex + 1;
+        }else if(signToPrimitiveMap.containsKey(currChar)){
+            sb.append(signToPrimitiveMap.get(currChar));
+            return textIndex+1;
+        }
+        return -1;
+    }
+
+    private static int parseArgs(StringBuilder sb, String text, int textIndex) {
+        if(textIndex + 1 >= text.length()) return -1;
+        if(text.charAt(textIndex) == ' ' && text.charAt(textIndex+1) == '(') {
+            sb.append(" (");
+            textIndex += 2;
+            boolean argsFound = false;
+            int i = textIndex;
+
+            while (i < text.length() && text.charAt(i) != ')' && (i = parseTerminalType(sb, text, textIndex)) != -1) {
+                sb.append(',');
+                textIndex = i;
+                argsFound = true;
+            }
+            if(i >= text.length()) return -1;
+            if (text.charAt(textIndex) == ')') {
+                textIndex += 1;
+                if (argsFound) {
+                    sb.setCharAt(sb.length() - 1, ')');
+                } else {
+                    sb.append(')');
+                }
+                return textIndex;
+            }
+        }
+        return -1;
+    }
+
+    private static int parseMethodName(StringBuilder sb, String text, int textIndex) {
+        if(textIndex >= text.length()) return -1;
+        if(text.charAt(textIndex) == '#') {
+            sb.append(".");
+            int endIndex = text.indexOf(" ");
+            if(endIndex == -1) return -1;
+            sb.append(text, textIndex+1, endIndex);
+            return endIndex;
+        }
+        return -1;
+    }
+
+    private static int parseRet(StringBuilder sb, String text, int textIndex) {
```

Comment:
Will remove the method, sure we can use the generic parseType for the parsing the return type.